pyro/infer/trace_mean_field_elbo.py

```diff
@@ -106,7 +106,6 @@ def _differentiable_loss_particle(self, model_trace, guide_trace):
                         kl_qp = kl_divergence(guide_site["fn"], model_site["fn"])
                         kl_qp = scale_and_mask(kl_qp, scale=guide_site["scale"], mask=guide_site["mask"])
                         if torch.is_tensor(kl_qp):
-                            assert kl_qp.shape == guide_site["fn"].batch_shape
```

Comment:
Rather than remove the assertion, could you gate by `torch._C.get_tracing_state()` as we do elsewhere in Pyro?
```py
if not torch._C.get_tracing_state():
    assert kl_qp.shape == guide_site["fn"].batch_shape
```